native: Dggsvp3 incorrectly calculates range for A cleanup

If a test is run with m=3, p=3, n=5 the code at [line 264](https://github.com/gonum/lapack/blob/master/native/dggsvp3.go#L264) panics with a bounds check failure.

```
--- FAIL: TestDggsvp3 (0.00s)
panic: runtime error: slice bounds out of range [recovered]
	panic: runtime error: slice bounds out of range

goroutine 5 [running]:
testing.tRunner.func1(0xc4200724e0)
	/home/daniel/Development/go/src/testing/testing.go:622 +0x29d
panic(0x6a1fc0, 0x895c00)
	/home/daniel/Development/go/src/runtime/panic.go:489 +0x2cf
github.com/gonum/lapack/native.Implementation.Dggsvp3(0x515655, 0x3, 0x3, 0x5, 0xc420080980, 0xf, 0xf, 0x5, 0xc420080a80, 0xf, ...)
	/home/daniel/Development/src/github.com/gonum/lapack/native/dggsvp3.go:264 +0xcf0
github.com/gonum/lapack/native.(*Implementation).Dggsvp3(0x8b84e8, 0x515655, 0x3, 0x3, 0x5, 0xc420080980, 0xf, 0xf, 0x5, 0xc420080a80, ...)
	<autogenerated>:25 +0x28b
github.com/gonum/lapack/testlapack.Dggsvp3Test(0xc4200724e0, 0x857340, 0x8b84e8)
	/home/daniel/Development/src/github.com/gonum/lapack/testlapack/dggsvp3.go:109 +0xd20
github.com/gonum/lapack/native.TestDggsvp3(0xc4200724e0)
	/home/daniel/Development/src/github.com/gonum/lapack/native/lapack_test.go:116 +0x5b
testing.tRunner(0xc4200724e0, 0x6e5600)
	/home/daniel/Development/go/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/home/daniel/Development/go/src/testing/testing.go:697 +0x2ca
FAIL	github.com/gonum/lapack/native	0.005s
```

--- a/native/dggsvp3.go
+++ b/native/dggsvp3.go
@@ -260,8 +260,8 @@ func (impl Implementation) Dggsvp3(jobU, jobV, jobQ lapack.GSVDJob, m, p, n int,
 		}
 
 		// Clean up A.
-		for i := k + 1; i < k+l; i++ {
-			r := a[i*lda+n-l : i*lda+i-k+n-l]
+		for i := k + 1; i < m; i++ {
+			r := a[i*lda+n-l : i*lda+min(n-l+i-k, n)]
 			for j := range r {
 				r[j] = 0
 			}

--- a/testlapack/dggsvd3.go
+++ b/testlapack/dggsvd3.go
@@ -25,6 +25,7 @@ func Dggsvd3Test(t *testing.T, impl Dggsvd3er) {
 
 		ok bool
 	}{
+		{m: 3, p: 3, n: 5, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0, ok: true},
 		{m: 5, p: 5, n: 5, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0, ok: true},
 		{m: 5, p: 5, n: 5, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0, ok: true},
 		{m: 5, p: 5, n: 10, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0, ok: true},

--- a/testlapack/dggsvp3.go
+++ b/testlapack/dggsvp3.go
@@ -23,6 +23,7 @@ func Dggsvp3Test(t *testing.T, impl Dggsvp3er) {
 	for cas, test := range []struct {
 		m, p, n, lda, ldb, ldu, ldv, ldq int
 	}{
+		{m: 3, p: 3, n: 5, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0},
 		{m: 5, p: 5, n: 5, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0},
 		{m: 5, p: 5, n: 5, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0},
 		{m: 5, p: 5, n: 10, lda: 0, ldb: 0, ldu: 0, ldv: 0, ldq: 0},

